src/Dfc.CourseDirectory.WebV2/Features/Venues/AddVenue/PostcodeSearch.cs

```diff
@@ -0,0 +1,194 @@
+                var vm = await CreateViewModel(request.Postcode);
+                return new ModelWithErrors<SearchViewModel>(vm, validationResult);
+            }
+
+            var addressDetail = await _addressSearchService.GetById(request.AddressId);
+
+            if (addressDetail == null)
+            {
+                // Address specified isn't valid. (We should never get here.)
+                // Re-run the search to get a list of valid addresses with a prompt to select an address.
+
+                var badAddressValidationResult = new ValidationResult(new[]
+                {
+                    new ValidationFailure(nameof(SelectCommand.AddressId), "Select an address")
+                });
+
+                var vm = await CreateViewModel(request.Postcode);
+                return new ModelWithErrors<SearchViewModel>(vm, badAddressValidationResult);
+            }
+
+            // SearchCommand has already checked that there is an ONSPD entry for this postcode
+            // so there is no valid case where `.Single()` should blow up
+            var onspdSearchResult = await _onspdSearchClient.Search(new OnspdSearchQuery() { Postcode = addressDetail.Postcode });
+            var onspdPostcodeRecord = onspdSearchResult.Items.Single();
+
+            _journeyInstance.UpdateState(state =>
+            {
+                state.AddressLine1 = addressDetail.Line1;
+                state.AddressLine2 = addressDetail.Line2;
+                state.Town = addressDetail.PostTown;
+                state.County = addressDetail.County;
+                state.Postcode = addressDetail.Postcode;
+                state.Latitude = onspdPostcodeRecord.Record.lat;
+                state.Longitude = onspdPostcodeRecord.Record.@long;
+                state.AddressIsOutsideOfEngland = onspdPostcodeRecord.Record.Country != "England";
+            });
+
+            return new Success();
+        }
+
+        private async Task<SearchViewModel> CreateViewModel(string postcode)
+        {
+            var results = await _addressSearchService.SearchByPostcode(postcode);
+
+            return new SearchViewModel()
+            {
+                Postcode = postcode,
+                Results = results
+                    .Select(r => new SearchResult()
+                    {
+                        AddressId = r.Id,
+                        StreetAddress = r.StreetAddress
+                    })
+                    .ToArray()
+            };
+        }
+
+        private void ThrowIfFlowStateNotValid()
+        {
+            _journeyInstance.ThrowIfCompleted();
+        }
+
+        private class SearchCommandValidator : AbstractValidator<SearchCommand>
+        {
+            public SearchCommandValidator()
+            {
+                RuleFor(c => c.Postcode)
+                    .NotEmpty().WithMessage("Enter a postcode")
+                    .Apply(Rules.Postcode).WithMessage("Enter a real postcode");
```

Comment:
"Enter a _real_ postcode"? Surely "Enter a _valid_ postcode" might be more appropriate?

Reply:
I didn't write the content

Reply:
I can't find a reference to that copy anywhere on the ticket or in the prototype - could you point me to the right place please?